src/octoprint/plugins/softwareupdate/__init__.py

```diff
@@ -1331,6 +1328,20 @@ def perform_update(self):
 
         force = json_data.get("force", "false") in octoprint.settings.valid_boolean_trues
 
+        if self._printer.is_printing() or self._printer.is_paused():
+            # do not update while a print job is running
+            # store targets to be run later on print done event
+            if self._queued_updates is not None:
+                self._queued_updates = dict_merge(
+                    self._queued_updates, {"targets": targets, "force": force}
+                )
```

Comment:
Thanks, that's my misunderstanding of how dict_merge worked relative to lists. Interesting during my testing I didn't have any issues but can't remember if I tried more than just one at a time. Will adjust.

Reply:
slightly different approach because your method would allow duplicates.

```
            if self._queued_updates is not None:
                self._queued_updates["targets"] = list(
                    set(self._queued_updates["targets"] + targets)
                )
            else:
                self._queued_updates = {"targets": targets, "force": force}
```